src/main/java/com/google/starfish/services/FavoriteNoteService.java

```diff
@@ -0,0 +1,152 @@
+package com.google.starfish.services;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;  
+import java.sql.ResultSet;  
+import javax.sql.DataSource;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Date;
+import com.google.starfish.models.Note;
+
+/**
+ * Service class for FavoriteNotes Table
+ */
+public class FavoriteNoteService {
+
+  /** Gets a favorite note by the compoud id */
+  public ResultSet getRowByCompoundId(DataSource pool, long noteId, String userId) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "SELECT * "
+        + "FROM favorite_notes "
+        + "WHERE `note_id`= ? AND "
+        + "`user_id` = ? "
+        + "LIMIT 1;";
+      try (PreparedStatement getStmt = conn.prepareStatement(stmt)) {
+        getStmt.setLong(1, noteId);
+        getStmt.setString(2, userId);
+        ResultSet rs = getStmt.executeQuery();
+        return rs;
+      } 
+    }
+  }
+
+  /** Deletes a favorite note by the compound id */
+  public boolean deleteRowByCompoundId(DataSource pool, long noteId, String userId) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "DELETE * "
+        + "FROM favorite_notes "
+        + "WHERE `note_id`= ? AND "
+        + "`user_id` = ? "
+        + "LIMIT 1;";
+      try (PreparedStatement deleteStmt = conn.prepareStatement(stmt)) {
+        deleteStmt.setLong(1, noteId);
+        deleteStmt.setString(2, userId);
+        boolean deleted = deleteStmt.execute();
+        return deleted;
+      } 
+    }
+  }
+
+  /** Inserts a favorite note by noteId and userId */
+  public void insertFavoriteNote(DataSource pool, long noteId, String userId) {
+    if (userId == null) return;
+    try (Connection conn = pool.getConnection()) {
+      try {
+        conn.setAutoCommit(false);
+        String stmt =
+            "INSERT INTO favorite_notes ( "
+                + "note_id,"
+                + "user_id ) "
+          + "VALUES ( "
+                + "?,"
+                + "? ); ";
+        try (PreparedStatement insertStmt = conn.prepareStatement(stmt)) {
+          insertStmt.setLong(1, noteId);
+          insertStmt.setString(2, userId);
+          System.out.println(insertStmt);
```

Comment:
nit: Do you still need the print statement here?

Reply:
Nope we don't, I'll remove!